spec/core/v2/ics-004-packet-semantics/PACKET.md

```diff
@@ -0,0 +1,116 @@
+# Packet Specification
+
+## Packet V2
+
+The IBC packet sends application data from a source chain to a destination chain with a timeout that specifies when the packet is no longer valid. The packet will be committed to by the source chain as specified in the ICS-24 specification. The receiver chain will then verify the packet commitment under the ICS-24 specified packet commitment path. If the proof succeeds, the IBC handler sends the application data(s) to the relevant application(s).
+
+```typescript
+interface Packet {
+    // identifier on the source chain
+    // that source chain uses to address dest chain
+    // this functions as the sending address
+    sourceIdentifier: bytes,
```

Comment:
Once two chains connected, by providing the `destIdentifier`, the IBC core should be able to retrieve the `sourceIdentifier` from storage and determine its value. Isn’t requiring this info unnecessary? After all, the `sourceIdentifier` linked to a `destIdentifier` will have to be stored anyway so that can perform a sanity check.

Reply:
On the source chain there is a map from `sourceId -> destId` and vice versa.

But the source chain does not have a map `destId -> sourceId` because a counterparty's id for us is not unique from our perspective. So we do need the sourceId in the packet **if** we are going to use it on acknowledgement/timeout to retrieve the correct information

Reply:
Are the source and destination identifiers supposed to be used like client ID or channel ID? If there are two separate IBC clients A1 and A2 created on Chain A for Chain B, can they both be used to send packets to the same IBC client B on chain B?

My feeling is that this should be possible, as otherwise we would need to perform redundant client updates. But to allow this, either we have to introduce something like channel ID as an indirection, or the chain should use both source and destination identifiers to form a unique pairing, and use it for nonce verification and sandboxing.